src/Lean/Elab/Notation.lean

```diff
@@ -55,13 +55,13 @@ def mkSimpleDelab (attrKind : Syntax) (vars : Array Syntax) (pat qrhs : Syntax)
     -- replace head constant with (unused) antiquotation so we're not dependent on the exact pretty printing of the head
     `(@[$attrKind:attrKind appUnexpander $(mkIdent c):ident]
       aux_def unexpand $(mkIdent c) : Lean.PrettyPrinter.Unexpander := fun
-       | `($$(_):ident $args*) => `($pat)
-       | _                     => throw ())
+       | `($$tk:ident $args*) => withRef tk `($pat)
```

Comment:
It's not really a token, is it? And could you add an explanatory comment? It is not clear to me who actually consumes that ref.